js_modules/dagit/packages/core/src/assets/AssetNodeLineageGraph.tsx

```diff
@@ -36,7 +37,8 @@ export const AssetNodeLineageGraph: React.FC<{
 
   React.useEffect(() => {
     if (viewportEl.current && layout) {
-      viewportEl.current.autocenter(false);
+      const lastZoomLevel = Number(window.localStorage.getItem(LINEAGE_GRAPH_ZOOM_LEVEL));
```

Comment:
Use `useStateWithStorage`?

Reply:
Hmm I tried to do this just now, but this is a bit of a weird case where we don't want the setter to trigger a re-render, because the scale is actually not a controlled prop at this level. I realized we do need to try catch this though so i'll make a few edits and re-use the helpers in useStateWithStorage